File: goldilocks/java/point.java
class Point {
    public int x;
    public int y;
}

class X {
    static void main() {
        var p = new Point();
        foo(p);
        // What's the value of p?
    }
}

class NumberStack {
    ArrayList<float> list = new ArrayList<float>;
    
    public boolean empty();
    public float peek() throws EmptyStackException;
    public float pop() throws EmptyStackException;
    public void push(float);
}

File: goldilocks/typescript/point.ts
interface ImmutablePoint { readonly x: number; readonly y: number; }

class Point {
    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }
    x: number;
    y: number;
}

function bar(p: Point): number {
    return 2;
}

function foo() {
    const p = new Point(1, 2);
    let x = bar(p);
    // What's the value of p now?
    x += 1;
}